src/json.cc

```diff
@@ -224,7 +226,35 @@ Obj JSON_ESCAPE_STRING(Obj self, Obj param)
     return copy;
 }
 
+Obj GAP_PLIST_TO_JSON_STRING(Obj self, Obj string, Obj stream, Obj list) {
+    Int len = LEN_PLIST(list);
```

Comment:
A personal thought (feel free to ignore): I really don't like having kernel functions that perform no argument checking. A test for `IS_PLIST` is cheap, and even expert users sometimes mix up arguments and getting weird crashes or segfaults is annoying. Helpers like `RequirePlainList` and `RequireStringRep` are available since GAP 4.11.1 (the latter calls `ConvString` if needed). If compatibility with older GAP versions is desirable, a simple `if (!IS_PLIST(list)) ErrorMayQuit("not a plist",0,0);` is still better than no check at all -- IMHO at least